src/shogun/mathematics/linalg/LinalgBackendBase.h

```diff
@@ -76,59 +77,124 @@ class LinalgBackendBase
 	BACKEND_GENERIC_DOT(complex128_t);
 	#undef BACKEND_GENERIC_DOT
 
+	/** @see linalg::sum */
+	#define BACKEND_GENERIC_SUM(Type, Class) \
+	virtual Type sum(const Class<Type>&) const \
+	{  \
+		SG_SNOTIMPLEMENTED; \
+	}
+
+	BACKEND_GENERIC_SUM(bool, SGVector);
+	BACKEND_GENERIC_SUM(char, SGVector);
+	BACKEND_GENERIC_SUM(int8_t, SGVector);
+	BACKEND_GENERIC_SUM(uint8_t, SGVector);
+	BACKEND_GENERIC_SUM(int16_t, SGVector);
+	BACKEND_GENERIC_SUM(uint16_t, SGVector);
+	BACKEND_GENERIC_SUM(int32_t, SGVector);
+	BACKEND_GENERIC_SUM(uint32_t, SGVector);
+	BACKEND_GENERIC_SUM(int64_t, SGVector);
+	BACKEND_GENERIC_SUM(uint64_t, SGVector);
+	BACKEND_GENERIC_SUM(float32_t, SGVector);
+	BACKEND_GENERIC_SUM(float64_t, SGVector);
+	BACKEND_GENERIC_SUM(floatmax_t, SGVector);
+	BACKEND_GENERIC_SUM(complex128_t, SGVector);
+	BACKEND_GENERIC_SUM(bool, SGMatrix);
+	BACKEND_GENERIC_SUM(char, SGMatrix);
+	BACKEND_GENERIC_SUM(int8_t, SGMatrix);
+	BACKEND_GENERIC_SUM(uint8_t, SGMatrix);
+	BACKEND_GENERIC_SUM(int16_t, SGMatrix);
+	BACKEND_GENERIC_SUM(uint16_t, SGMatrix);
+	BACKEND_GENERIC_SUM(int32_t, SGMatrix);
+	BACKEND_GENERIC_SUM(uint32_t, SGMatrix);
+	BACKEND_GENERIC_SUM(int64_t, SGMatrix);
+	BACKEND_GENERIC_SUM(uint64_t, SGMatrix);
+	BACKEND_GENERIC_SUM(float32_t, SGMatrix);
+	BACKEND_GENERIC_SUM(float64_t, SGMatrix);
+	BACKEND_GENERIC_SUM(floatmax_t, SGMatrix);
+	BACKEND_GENERIC_SUM(complex128_t, SGMatrix);
+	#undef BACKEND_GENERIC_SUM
```

Comment:
btw if you think that this is a better idea, can you please make your existing PRs work with this meta-macro instead? most of your PRs look fine. we can merge it then. 

Maybe add this to one of your PRs, let me or @karlnapf know that you want that to be merged first, get it merged, rebase your other PRs against it? sounds okay?